### PR TITLE
Add hand class

### PR DESCRIPTION
**Pull request by Sebdevar**
I recommend going through the request, one commit at a time so that you can understand the process correctly:

Basically:
1. Your create the skeleton of what you want without implementing anything.
2. Your write your base tests, then slowly think of limit tests (as you go through, you'll see the limit cases appearing)
3. Run tests, most will fail, others will be false positives
4. Implement the class while paying attention to your tests, bringing each of them green one by one until finally you're done.

This is a really basic case off TDD and although the class may look really small and basic and unrequiring of tests... it's by writing the tests that you end up figuring out how to make it as simple as it becomes at the end.

TDD is hard process to start and I won't ask you to do it, but there is a huge value in it.

### CODE REVIEW

**Review by AshaGakunin — CHANGES_REQUESTED** [review]
For a base hand class after reading the rules of baccarat you have caught all cases that i could think of.
You checked when face cards are added that the value doesn't change, you ensured the number of cards can't exceed 3 when adding cards.  
Maybe add a boolean variable FullHand for ease in checking later in the final stage of the game round.  Granted You could check do something like "if(dealerHand.cards.length==3 && playerHand.cards.Length==3){...code to continue with deciding winner}"  however i think dealerHand.FullHand as a boolean would be cleaner and avoid the .length method.

I am nitpicking but i am trying to do my best to take this seriously.

So in addition to the FullHand Boolean, tests should be added that test if a hands.FullHand Boolean switches after adding 3 cards.

**Comment by Sebdevar** [comment]
That is definitely not nitpicking and I couldn't be prouder of you for bringing this up.

I will make that change tomorrow and push it, so you'll be able to go through the entire process of a professional code review, by going over it before approving.

The cool thing about branches is that this won't keep me from continuing to work on a separate portion via another branch while this one is waiting for approval.

Thanks a ton for your help!

**Review by AshaGakunin — APPROVED** [review]

**Pull request merged**
